packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -773,6 +773,16 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         case tt.bracketL:
           return this.tsParseTupleType();
         case tt.parenL:
+          if (!this.options.createParenthesizedExpressions) {
+            const startPos = this.state.start;
+            this.expect(tt.parenL);
```

Comment:
It suffices to move the cursor only `this.next()` instead of `this.expect(tt.parenL)` because the switch case has implied `this.match(tt.parenL)`.